src/metabase/query_processor/middleware/permissions.clj

```diff
@@ -89,16 +90,29 @@
   (doseq [{query :dataset_query} (qp.u.tag-referenced-cards/tags-referenced-cards outer-query)]
     (check-query-permissions* query)))
 
-(s/defn ^:private check-query-permissions*
+(def ^:dynamic *user-can-read-dashboard*
```

Comment:
Maybe this could be named something like `*param-values-query*` to make it very specific what the use case of this var is?